File: java/duck/ducks/RedheadDuck.java
package ducks;

import behaviors.fly.FlyWithWings;
import behaviors.quack.Squeak;

public class RedheadDuck extends Duck {

    public RedheadDuck(String name) {
        super(name);
        flyBehavior = new FlyWithWings(name);
        quackBehavior = new Squeak(name);
    }
}

File: kotlin/duck/ducks/RedheadDuck.kt
package ducks

import behaviors.fly.FlyWithWings
import behaviors.quack.Squeak

class RedheadDuck(name: String) : Duck(name) {

    init {
        quackBehavior = Squeak(name)
        flyBehavior = FlyWithWings(name)
    }

}